src/bluesearch/embedding_models.py

```diff
@@ -38,10 +39,10 @@ class EmbeddingModel(ABC):
 
     @property
     @abstractmethod
-    def dim(self):
+    def dim(self) -> int:
         """Return dimension of the embedding."""
 
-    def preprocess(self, raw_sentence):
+    def preprocess(self, raw_sentence: str) -> str:
```

Comment:
Maybe you can the opportunity to:
- Remove the typing from the Parameters section of the docstring (as it is a duplicate of the signature)
- Add the type of the Returns section

Reply:
done